spacy/tests/matcher/test_matcher_api.py

```diff
@@ -458,3 +458,11 @@ def test_matcher_callback(en_vocab):
     doc = Doc(en_vocab, words=["This", "is", "a", "test", "."])
     matches = matcher(doc)
     mock.assert_called_once_with(matcher, doc, 0, matches)
+
+
+def test_matcher_span(matcher):
+    text = "JavaScript is good but Python is better"
+    doc = Doc(matcher.vocab, words=text.split())
+    span = Span(doc, 0, 3)
+    matches = matcher(span)
+    assert len(matches) == 1
```

Comment:
I wonder whether we shouldn't just do `matcher(span.as_doc())` to support this use-case, then we wouldn't need to accomodate for `Span`'s specifically throughout the `Matcher` code?

Also, as a minor comment, this unit test would make more sense if the full `doc` would match more than once, e.g. by changing the sentence into "JavaScript is good but Java is better" --> full `doc` matches twice, the `span` only once.

Reply:
I see your point. Using `span.as_doc()` is a good solution for sure, but when scaling on the number of spans that is a heavier operation than using the Span created from slicing directly. I also feel a little redundancy in a use-case like matching really many spans from the same `Doc`.
Anyway, I'll adapt my use-cases to the current `matcher` if it's not seen a real need for this kind of feature.

I'll fix the tests in any case, thanks for your hint.

Reply:
Yep, I agree with the efficiency argument!